File: TransCoder_tokenized_test_set_functions/cpp/NUMBER_NODES_TWO_VERTICES_ACYCLIC_GRAPH_DISJOINT_UNION_METHOD.cpp
int totalNodes ( vector < int > adjac [ ], int n, int x, int y ) {
  bool visited [ n + 1 ] = {
    0 };
    int p [ n ];
    queue < int > q;
    q . push ( x );
    visited [ x ] = true;
    int m;
    while ( ! q . empty ( ) ) {
      m = q . front ( );
      q . pop ( );
      for ( int i = 0;
      i < adjac [ m ] . size ( );
      ++ i ) {
        int h = adjac [ m ] [ i ];
        if ( ! visited [ h ] ) {
          visited [ h ] = true;
          p [ h ] = m;
          q . push ( h );
        }
      }
    }
    int count = 0;
    int i = p [ y ];
    while ( i != x ) {
      count ++;
      i = p [ i ];
    }
    return count;
  }

File: TransCoder_tokenized_test_set_functions/python/NUMBER_NODES_TWO_VERTICES_ACYCLIC_GRAPH_DISJOINT_UNION_METHOD.py
def totalNodes ( adjac, n, x, y ) :
    visited = [ 0 ] * ( n + 1 )
    p = [ None ] * n
    q = queue . Queue ( )
    q . put ( x )
    visited [ x ] = True
    m = None
    while ( not q . empty ( ) ) :
        m = q . get ( )
        for i in range ( len ( adjac [ m ] ) ) :
            h = adjac [ m ] [ i ]
            if ( not visited [ h ] ) :
                visited [ h ] = True
                p [ h ] = m
                q . put ( h )
    count = 0
    i = p [ y ]
    while ( i != x ) :
        count += 1
        i = p [ i ]
    return count
